motor_controllers/kostal_inveor/firmware.lua

```diff
@@ -0,0 +1,126 @@
+local config = require('enapter.ucm.config')
+
+ADDRESS_CONFIG = 'address'
+BAUDRATE_CONFIG = 'baudrate'
+DATA_BITS_CONFIG = 'data_bits'
+STOP_BITS_CONFIG = 'stop_bits'
+PARITY_CONFIG = 'parity_bits'
+
+function main()
+  enapter.log('DEBUG MAIN')
+
+  config.init({
+    [ADDRESS_CONFIG] = { type = 'number', required = true, default = 1 },
+    [BAUDRATE_CONFIG] = { type = 'number', required = true, default = 9600 },
+    [DATA_BITS_CONFIG] = { type = 'number', required = true, default = 8 },
+    [STOP_BITS_CONFIG] = { type = 'number', required = true, default = 1 },
+    [PARITY_CONFIG] = { type = 'string', required = true, default = 'N' },
+  })
+
+  local values, err = config.read_all()
+  if err then
+    enapter.log('cannot read config: ' .. tostring(err), 'error')
+    return nil, 'cannot_read_config'
+  else
+    for _, value in pairs(values) do
+      if not value then
+        return nil, 'not_configured'
+      end
+    end
+  end
+
+  local baudrate = values[BAUDRATE_CONFIG]
+  local data_bits = values[DATA_BITS_CONFIG]
+  local parity_bits = values[PARITY_CONFIG]
+  local stop_bits = values[STOP_BITS_CONFIG]
+
+  local result = rs485.init(baudrate, data_bits, parity_bits, stop_bits)
+  if result ~= 0 then
+    enapter.log('RS-485 failed: ' .. result .. ' ' .. rs485.err_to_str(result), 'error', true)
+  end
+
+  scheduler.add(30000, send_properties)
+  scheduler.add(1000, send_telemetry)
+end
+
+function send_properties()
+  enapter.send_properties({ vendor = 'KOSTAL', model = 'INVEOR' })
+end
+
+function send_telemetry()
+  local telemetry = {}
+  local status = 'ok'
+
+  local values, err = config.read_all()
+  if err then
+    enapter.log('cannot read config: ' .. tostring(err), 'error')
+    return nil, 'cannot_read_config'
+  else
+    for _, value in pairs(values) do
+      if not value then
+        return nil, 'not_configured'
+      end
+    end
+  end
+
+  local address = values[ADDRESS_CONFIG]
+
+  local data, result = modbus.read_inputs(address, 999, 1, 1000)
+  if data then
+    telemetry['actual_freq'] = tofloat(data)
```

Comment:
`tofloat` function won't work in this script because it requires 2 registers, but you read only 1

Reply:
What should I do with this?